Assets/LeapC/LeapQuaternion.cs

```diff
@@ -0,0 +1,204 @@
+     * \include Vector_Equals.txt
+     * @since 1.0
+     */
+    public bool Equals(LeapQuaternion v)
+    {
+      return x.NearlyEquals(v.x) && y.NearlyEquals(v.y) && z.NearlyEquals(v.z) && w.NearlyEquals(v.w);
+    }
+    public override bool Equals(Object obj)
+    {
+      return obj is LeapQuaternion && Equals((LeapQuaternion)obj);
+    }
+
+    /**
+     * Returns true if all of the quaternion's components are finite.  If any
+     * component is NaN or infinite, then this returns false.
+     *
+     * \include Vector_IsValid.txt
+     * @since 1.0
+     */
+    public bool IsValid()
+    {
+      return float.IsNaN(x) && float.IsNaN(y) && float.IsNaN(z) && float.IsNaN(w);
+    }
+
+    /**
+     * The x component.
+     * @since 1.0
+     */
+    public float x;
+
+    /**
+     * The y component.
+     * @since 1.0
+     */
+    public float y;
+
+    /**
+     * The z component.
+     * @since 1.0
+     */
+    public float z;
+
+    /**
+     * The w component.
+     * @since 1.0
+     */
+    public float w;
+
+    /**
+     * The magnitude, or length, of this quaternion.
+     *
+     * The magnitude is the L2 norm, or Euclidean distance between the origin and
+     * the point represented by the (x, y, z) components of this LeapQuaternion object.
+     *
+     * \include Vector_Magnitude.txt
+     *
+     * @returns The length of this quaternion.
+     * @since 1.0
+     */
+    public float Magnitude
+    {
+      get { return (float)Math.Sqrt(x * x + y * y + z * z + w * w); }
+    }
+
+    /**
+     * The square of the magnitude, or length, of this quaternion.
+     *
+     * \include Vector_Magnitude_Squared.txt
+     *
+     * @returns The square of the length of this quaternion.
+     * @since 1.0
+     */
+    public float MagnitudeSquared
+    {
+      get { return x * x + y * y + z * z + w * w; }
+    }
+
+    /**
+     * A normalized copy of this quaternion.
+     *
+     * A normalized quaternion has the same direction as the original quaternion,
+     * but with a length of one.
+     *
+     * @returns A LeapQuaternion object with a length of one, with the same rotation
+     * direction as this LeapQuaternion object.
+     * @since 1.0
+     */
+    public LeapQuaternion Normalized
+    {
+      get
+      {
+        float denom = this.MagnitudeSquared;
+        if (denom <= Leap.Constants.EPSILON)
+        {
+          return LeapQuaternion.Identity;
+        }
+        denom = 1.0f / (float)Math.Sqrt(denom);
+        return new LeapQuaternion(x * denom, y * denom, z * denom, w * denom);
+      }
+    }
+
+    public LeapQuaternion Multiply(LeapQuaternion rhs)
+    {
+      return new LeapQuaternion(
+        w * rhs.x + x * rhs.w + y * rhs.z - z * rhs.y,
+        w * rhs.y + y * rhs.w + z * rhs.x - x * rhs.z,
+        w * rhs.z + z * rhs.w + x * rhs.y - y * rhs.x,
+        w * rhs.w - x * rhs.x - y * rhs.y - z * rhs.z);
+    }
+
+    public static readonly LeapQuaternion Identity = new LeapQuaternion(0, 0, 0, 1);
```

Comment:
I thought that a quaternion with (0,0,0) for the xyz value was invalid?  Shouldn't it be something like (1,0,0,0) ?